src/services/utilities.ts

```diff
@@ -1173,7 +1173,17 @@ namespace ts {
             }
 
             const children = n.getChildren(sourceFile);
-            for (let i = 0; i < children.length; i++) {
+            const i = binarySearchKey(children, position, (_, i) => i, (middle, _) => {

```

Comment:
A comparer that ignores one argument is definitely not more readable than what you started with.

Reply:
So... that we call it a "comparer" is itself a mistake on our part. Yes, you _can_ pass a comparer and it does what you might want, but we reliably pass the second argument to `binarySearchKey` as the second argument to the "comparer" - it never changes. I could choose to not ignore it, but there's no reason to - it's the same constant `position` value I close over.

Reply:
So it's more of a "selector" callback - it just needs to return if you "select the pivot element", "select the left group", or "select the right group". The `Comparison` values are just mapped to that.

Reply:
I think a comment would be helpful here.